### PR TITLE
Have a user.json in the base directory

### PR DESCRIPTION
**Pull request by alexmorley**
To make this opts file visible & nudge people not to make changes to default.json?

### CODE REVIEW

**Comment by magland** [comment]
Are you suggesting to add a mountainlab.user.json to the repo? But then if
the user changes it there will be a conflict with the repo. I agree it
would be helpful -- maybe it should be generated with compilation. Then it
can be a bare bones, and the defaults in mountainlab.default.json would
still take effect.

**Comment by alexmorley** [comment]
Yeah I added it to the .gitignore but it might still conflict. Generating during compilation is probably smarter. Can just add `if [ ! -f mountainlab.user.json ]; then; echo {} > mountainlab.user.json; fi` to the `./compile_components.sh` ?

**Comment by wysota** [comment]
compile_components is not always used so I'm not sure this is a good idea. And `echo {} > mountainlab.user.json` will truncate the file if it already exists. Better use `touch mountainlab.user.json`.

However IMO it's better to make mountainlab.default.json readonly and use the editor that's waiting in the other pull request for inclusion into master.

**Comment by magland** [comment]
Yes, maybe:

mountainlab.user.json ]; then; cp devtools/mountainlab.user.json.template
mountainlab.user.json;

I know that mountainlab.user.json.template is a crazy name, but it is
hidden in devtools, so as to not cause confusion to user.

What do you think?

File: mountainlab.user.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (wysota) [inline mountainlab.user.json]: I don't think it is a good idea to have a file with such content in the repo.

**Comment by magland** [comment]
Let's try making it readonly and using the new config editor (which I have not tried yet, but will merge in now), as wysota suggests. See how that works.

**Comment by alexmorley** [comment]
closed by #125

**Comment by magland** [comment]
Witold, could you allow launching the config editor from the single command: mlconfig

Also, if mountainlab.user.json does not exist... please create it as "{}"
Ideally I'd like some more structure in there such as below, but not sure what is the best practice. Those two fields are relevant for most users.

Suggestions?

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

{
	"general":{ 
		"temporary_path":"/tmp"
	},
	"mountainprocess":{
	},
	"prv":{
		"local_search_paths":["examples"],
	},
	"kron":{
	}
}

**Comment by wysota** [comment]
> Witold, could you allow launching the config editor from the single command: mlconfig

Yes

> Also, if mountainlab.user.json does not exist... please create it as "{}"

Create and save it as such or just put braces into the editor if the file is empty?

> Ideally I'd like some more structure in there such as below, but not sure what is the best practice. Those two fields are relevant for most users.

IMO ideally the user should not edit JSON directly. With the property browser I am using for render options available, we should be able to make it work for editing the config as well.

**Comment by magland** [comment]
On Sat, Feb 4, 2017 at 12:06 PM, Witold Wysota <notifications@github.com>
wrote:

> Witold, could you allow launching the config editor from the single
> command: mlconfig
>
> Yes
>
> Also, if mountainlab.user.json does not exist... please create it as "{}"
>
> Create and save it as such or just put braces into the editor if the file
> is empty?
>
I guess just put braces into the editor.



> Ideally I'd like some more structure in there such as below, but not sure
> what is the best practice. Those two fields are relevant for most users.
>
> IMO ideally the user should not edit JSON directly. With the property
> browser I am using for render options available, we should be able to make
> it work for editing the config as well.
>
Okay, sounds good.

**Pull request closed without merging**